src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -139,62 +193,114 @@ public function broadcast(array $channels, $event, array $payload = [])
+            $iat = $payload['iat'];
+            $exp = $payload['exp'];
+            $channelClaims = json_decode($payload['x-ably-capability'], true);
+        } else {
+            $iat = $serverTimeFn();
+            $exp = $iat + $this->tokenExpiry;
+        }
+        if ($channelName) {
+            $channelClaims[$channelName] = $channelCapability;
+        }
+        $claims = [
+            'iat' => $iat,
+            'exp' => $exp,
+            'x-ably-clientId' => $clientId ? strval($clientId) : null,
+            'x-ably-capability' => json_encode($channelClaims),
+        ];
+
+        return Utils::generateJwt($header, $claims, $this->getPrivateToken());
+    }
+
+    /**
+     * Remove prefix from channel name.
      *
-     * @param  string  $event
-     * @param  array  $payload
-     * @return \Ably\Models\Message
+     * @param $channel string
+     * @return string
      */
-    protected function buildAblyMessage($event, array $payload = [])
+    public function normalizeChannelName($channel)
     {
-        return tap(new AblyMessage, function ($message) use ($event, $payload) {
-            $message->name = $event;
-            $message->data = $payload;
-            $message->connectionKey = data_get($payload, 'socket');
-        });
+        if ($channel) {
+            if ($this->isPrivateChannel($channel)) {
+                return Str::replaceFirst('private:', '', $channel);
+            }
+            if ($this->isPresenceChannel($channel)) {
+                return Str::replaceFirst('presence:', '', $channel);
+            }
+
+            return Str::replaceFirst('public:', '', $channel);
+        }
+
+        return $channel;
     }
 
     /**
-     * Return true if the channel is protected by authentication.
+     * Checks if channel is a private channel.
      *
-     * @param  string  $channel
+     * @param $channel string
      * @return bool
      */
-    public function isGuardedChannel($channel)
+    public function isPrivateChannel($channel)
     {
-        return Str::startsWith($channel, ['private-', 'presence-']);
+        return Str::startsWith($channel, 'private:');
     }
 
     /**
-     * Remove prefix from channel name.
+     * Checks if channel is a presence channel.
      *
-     * @param  string  $channel
-     * @return string
+     * @param $channel string
+     * @return bool
      */
-    public function normalizeChannelName($channel)
+    public function isPresenceChannel($channel)
     {
-        if ($this->isGuardedChannel($channel)) {
-            return str_starts_with($channel, 'private-')
-                        ? Str::replaceFirst('private-', '', $channel)
-                        : Str::replaceFirst('presence-', '', $channel);
-        }
+        return Str::startsWith($channel, 'presence:');
+    }
 
-        return $channel;
+    /**
+     * Checks if channel needs authentication.
+     *
+     * @param $channel string
```

Comment:
```suggestion
     * @param  string  $channel
```
First variable type and then the variable name

Reply:
Fixed!